common/persistence/executionStore.go

```diff
@@ -295,13 +295,16 @@ func (m *executionManagerImpl) ConflictResolveWorkflowExecution(
 	if err != nil {
 		return err
 	}
+
+	if request.CurrentWorkflowMutation == nil {
+		return serviceerror.NewInternal("ConflictResolveWorkflowExecution: current workflow mutation not set on request")
+	}
 	var serializedCurrentWorkflowMutation *InternalWorkflowMutation
-	if request.CurrentWorkflowMutation != nil {
-		serializedCurrentWorkflowMutation, err = m.SerializeWorkflowMutation(request.CurrentWorkflowMutation)
-		if err != nil {
-			return err
-		}
+	serializedCurrentWorkflowMutation, err = m.SerializeWorkflowMutation(request.CurrentWorkflowMutation)
+	if err != nil {
+		return err
 	}
+
```

Comment:
this block need to be reverted
current workflow can be left intact
https://github.com/temporalio/temporal/blob/f286feb25ef9eef403533a0082ab377d1a81d54f/service/history/nDCTransactionMgrForExistingWorkflow.go#L448-L457